### PR TITLE
fix: json parse nan/inf for altair charts

### LINKED ISSUE #4568: JSON.parse error when plotting NaN with polars + Altair
### Describe the bug

When plotting a dataframe containing NaNs using polars + Altair, I get the following error:
```
Something went wrong

JSON.parse: unexpected character at line 16 column 14 of the JSON data

If this is an issue with marimo, please report it on [GitHub](https://github.com/marimo-team/marimo/issues).
```
If I replace the NaN in the example with an integer, the plot displays fine. I haven't figured out how to inspect the JSON data, but I presume the NaN is getting outputted oddly.

### Environment

<details>

```
{
  "marimo": "0.12.10",
  "OS": "Linux",
  "OS Version": "5.14.0-427.24.1.el9_4.x86_64",
  "Processor": "x86_64",
  "Python Version": "3.12.3",
  "Binaries": {
    "Browser": "135.0.7049.95",
    "Node": "v16.20.2"
  },
  "Dependencies": {
    "click": "8.1.8",
    "docutils": "0.18.1",
    "itsdangerous": "2.2.0",
    "jedi": "0.19.2",
    "markdown": "3.6",
    "narwhals": "1.32.0",
    "packaging": "24.2",
    "psutil": "7.0.0",
    "pygments": "2.19.1",
    "pymdown-extensions": "10.14.3",
    "pyyaml": "6.0.2",
    "starlette": "0.46.2",
    "tomlkit": "0.13.2",
    "typing-extensions": "4.12.2",
    "uvicorn": "0.34.0",
    "websockets": "15.0.1"
  },
  "Optional Dependencies": {
    "altair": "5.5.0",
    "nbformat": "5.10.4",
    "pandas": "2.2.3",
    "polars": "1.26.0",
    "pyarrow": "19.0.1",
    "pycrdt": "0.11.1",
    "pytest": "8.3.5",
    "python-lsp-ruff": "2.2.2",
    "python-lsp-server": "1.12.2",
    "ruff": "0.9.6"
  },
  "Experimental Flags": {
    "lsp": true
  }
}

```

</details>

### Code to reproduce

```python
import polars as pl
import numpy as np

pl.DataFrame({"x": [1,2,3], "y": [np.nan, 3, 4]}).plot.line(x="x", y="y")
```

### PR DESCRIPTION
**Pull request by Light2Dark**
## 📝 Summary

Fixes #4568 .

we could also use special json parse, but this fixes the issue at the source.

I've also tested and found that these values don't affect how a chart is displayed, it should be converted to null. If you run this in jupyter, they are also converted to null.

## 📋 Checklist

- [X] I have read the [contributor guidelines](https://github.com/marimo-team/marimo/blob/main/CONTRIBUTING.md).
- [ ] For large changes, or changes that affect the public API: this change was discussed or approved through an issue, on [Discord](https://marimo.io/discord?ref=pr), or the community [discussions](https://github.com/marimo-team/marimo/discussions) (Please provide a link if applicable).
- [X] I have added tests for the changes made.
- [X] I have run the code and verified that it works as expected.

## 📜 Reviewers

@mscolnick

### CODE REVIEW

File: marimo/_output/formatters/altair_formatters.py [hunk 1]
```diff
@@ -118,3 +121,23 @@ def _apply_embed_options(chart: altair.Chart) -> altair.Chart:
...
+    This function is a wrapper around the altair.Chart.to_json method.
+    It sanitizes the data in the chart if necessary and validates the spec.
+    """
+    try:
+        return chart.to_json(
+            format=spec_format, validate=validate, allow_nan=False
+        )
+    except ValueError:
+        chart.data = sanitize_nan_infs(chart.data)
+        return chart.to_json(format=spec_format, validate=validate)
```
Comment (mscolnick) [inline marimo/_output/formatters/altair_formatters.py:143]: What about to_dict and then json.dumps(allow_nan=False)

Comment (Light2Dark) [reply marimo/_output/formatters/altair_formatters.py:143]: chart.to_dict also causes `nan` or `inf` values
```json
[
  {
    "x": 1,
    "y": nan
  },
  {
    "x": 2,
    "y": 3.0
  },
  {
    "x": 3,
    "y": inf
  }
]
```

Comment (mscolnick) [reply marimo/_output/formatters/altair_formatters.py:143]: `to_json` also accepts `allow_nans=False`

Comment (Light2Dark) [reply marimo/_output/formatters/altair_formatters.py:143]: Ah, okay, `chart.to_json` [calls `json.dumps`](https://github.com/vega/altair/blob/807ffb5b825b4736621432ab4401472ee82e18bf/altair/vegalite/v5/api.py#L2047) under the hood. Passing [allow_nan=False triggers the ValueError](https://docs.python.org/3/library/json.html#basic-usage), it rejects invalid json.

Comment (mscolnick) [reply marimo/_output/formatters/altair_formatters.py:143]: oh i see. i misread the `json.dumps` spec. i just assumed it would remove nans instead of raise an error. thanks for explaining
